Update Dockerfile/pefile v.1.2.10-139

wzod opened the pull request:
Changelog:
- update Dockerfile to include recent changes in base repo
- update pefile to version 1.2.10-139; resolve issues when using version 1.2.10-114 (apt/pip repos are a bit behind)

I have pushed the changes to DockerHub; updating the Dockerfile for consistency/reference.

wzod committed:
Merge pull request #1 from lmco/master

Sync Forked Repo

wzod committed:
Update Dockerfile/pefile v.1.2.10-139 

Changelog:

- update Dockerfile to include recent changes in base repo
- update pefile to version 1.2.10-139; resolve issues when using version 1.2.10-114 (apt/pip repos are a bit behind)

wzod committed:
Revert

Revert to previous commit

wzod committed:
Update Dockerfile/pefile v.1.2.10-139

Changelog:

- update Dockerfile to include recent changes in base repo
- update pefile to version 1.2.10-139; resolve issues when using version 1.2.10-114 (apt/pip repos are a bit behind)

marnao commented:
apt is indeed a bit behind but pip has the latest tagged release at 1.2.10-139. I agree we should at least be using the latest version in pip as opposed to apt because META_PE relies on some of these changes.

However, your pulling master here which is even newer. I'm unclear if any of the changes in master are incompatible with META_PE. I tested new META_PE with 1.2.10-139 and had no issues. Have you done any testing with master?

Whatever we decide, we should update the README to reflect the change also.

wzod committed:
Update Dockerfile (use pypi)

Changelog:

- removed pefile download from github (master) and substituted with download URL from pypi

wzod committed:
Update README

Changelog:

- update README to remove install of python-pefile from apt repo and include instructions for downloading and installing pefile from pypi download URL

wzod commented:
Good point with not pulling from master since that adds considerable risk if/when new changes are introduced in the pefile repo. Granted in the case of the docker image, the build is from a snapshot in time as of when the image was created in DockerHub; thus, the docker image would continue to use what's currently in master (version 1.2.10-139). On the other hand, that may become problematic should someone else wish to build their own image or follow the steps in the Dockerfile to create their own instance.

As a workaround, I updated the pull request by removing the pull from master and substituting with the download URL from pypi ( http://pefile.googlecode.com/files/pefile-1.2.10-139.tar.gz ). That should provide a consistent means of grabbing pefile source without having to rely on what's contained in master.

I had tested pefile from master and had no issues when I made the change yesterday; likely since the code has not changed much and still appears to be at version 1.2.10-139.  I have also tested this most recent change with using pefile from pypi and have not had any issues.

Rationale for pulling from master was due to issues with installing from source after downloading from the releases page ( https://github.com/erocarrera/pefile/releases ). Thanks for pointing out pypi as another alternative as that seems to be the best option right now. 

On a separate note,I have submitted a request to update the releases page as that would be a convenient place to download the source files in the future ( https://github.com/erocarrera/pefile/issues/80 ).

Regarding README, I took a crack at updating the instructions by removing python-pefile from apt repo and including steps for downloading/installing from pypi.

wzod committed:
Use Consistent Location for Source Files 

Change log:

- minor edits to ensure all source files reside in /tmp directory

marnao commented:
Thanks for the modifications. I think using a release instead of master is the safest bet, even if the release is a bit long in the tooth. Breaking changes could appear in the future. Hopefully they tag a new release soon so we can test and switch to it.

Also thanks for updating the README to match!
